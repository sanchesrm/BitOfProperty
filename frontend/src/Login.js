import React, { Component } from "react";
import logo from "./logo.svg";
import "./App.css";
import { Form, Icon, Input, Button, Layout } from "antd";
import "antd/dist/antd.css";
import { withRouter } from "react-router-dom";

const { Content } = Layout;
const FormItem = Form.Item;

class LoginForm extends Component {
  handleSubmit = e => {
    e.preventDefault();
    this.props.form.validateFields((err, values) => {
      if (!err) {
        console.log("Received values of form: ", values);
        this.props.history.push("/mainPage");
      }
    });
  };

  render() {
    const { getFieldDecorator } = this.props.form;

    return (
      <Content style={{ padding: "10% 40%" }}>
        <Form onSubmit={this.handleSubmit} style={{ maxWidth: "300px" }}>
          <FormItem>
            {getFieldDecorator("userName", {
              rules: [{ required: true, message: "Please input your username!" }]
            })(<Input prefix={<Icon type="user" />} placeholder="Username" />)}
          </FormItem>
          <FormItem>
            {getFieldDecorator("password", {
              rules: [{ required: true, message: "Please input your Password!" }]
            })(<Input prefix={<Icon type="lock" />} type="password" placeholder="Password" />)}
          </FormItem>
          <FormItem>
            <Button type="primary" htmlType="submit" style={{ width: "100%" }}>
              Log in
            </Button>
          </FormItem>
        </Form>
      </Content>
    );
  }
}

const WrappedLoginForm = Form.create()(LoginForm);

export default withRouter(WrappedLoginForm);
