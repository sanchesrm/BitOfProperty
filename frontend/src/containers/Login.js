import React, { Component } from "react";
import { Form, Icon, Input, Button, Layout, message } from "antd";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { bindActionCreators, compose } from "redux";
import { signInUser } from "../actions/signinActions";
import "antd/dist/antd.css";
import logo from "../imgs/bitofproperty.png";

export class LoginForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      loading: false
    };
  }

  handleSubmit = e => {
    e.preventDefault();
    this.props.form.validateFields((err, values) => {
      if (!err) {
        this.setState({
          loading: true
        });

        this.props
          .signInUser(values)
          .then(() => {
            this.props.history.push("/");
          })
          .catch(() => {
            message.error(
              "Something went wrong! Check your username or password!",
              3
            );
            this.setState({
              loading: false
            });
          });
      }
    });
  };

  render() {
    const { getFieldDecorator } = this.props.form;

    return (
      <Layout.Content style={{ padding: "10% 40%" }}>
        <img src={logo} style={{ maxWidth: "300px" }} alt="" />

        <Form onSubmit={this.handleSubmit} style={{ maxWidth: "300px" }}>
          <Form.Item>
            {getFieldDecorator("username", {
              rules: [
                {
                  type: "email",
                  message: "Username should be an e-mail!"
                },
                {
                  required: true,
                  message: "Please put your username"
                }
              ]
            })(<Input prefix={<Icon type="user" />} placeholder="Username" />)}
          </Form.Item>
          <Form.Item>
            {getFieldDecorator("password", {
              rules: [
                { required: true, message: "Please input your Password!" }
              ]
            })(
              <Input
                prefix={<Icon type="lock" />}
                type="password"
                placeholder="Password"
              />
            )}
          </Form.Item>
          <Form.Item>
            <Button
              type="primary"
              htmlType="submit"
              style={{ width: "100%" }}
              loading={this.state.loading}
            >
              Log in
            </Button>
          </Form.Item>
        </Form>
      </Layout.Content>
    );
  }
}

export const WrappedLoginForm = Form.create()(LoginForm);

export const mapDispatchToProps = dispatch => {
  return bindActionCreators({ signInUser }, dispatch);
};

export default compose(
  withRouter,
  connect(
    null,
    mapDispatchToProps
  )
)(WrappedLoginForm);
