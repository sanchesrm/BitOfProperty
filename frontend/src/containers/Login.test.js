import Adapter from "enzyme-adapter-react-16";
import React from "react";
import toJson from "enzyme-to-json";
import { configure, mount } from "enzyme";
import { LoginForm, WrappedLoginForm } from "./Login";
import { Form } from "antd";

describe("Login Component", () => {
  let props;
  beforeEach(() => {
    configure({ adapter: new Adapter() });
    props = {
      signInUser: jest.fn(() => Promise.resolve()),
      history: {
        push: jest.fn()
      }
    };
  });

  test("render correctly", () => {
    expect.assertions(1);

    const EnhancedForm = Form.create()(WrappedLoginForm);
    const wrapper = mount(<EnhancedForm {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("should validate required fields", () => {
    expect.assertions(1);

    const EnhancedForm = Form.create()(WrappedLoginForm);
    const wrapper = mount(<EnhancedForm {...props} />);

    const signInUserMethod = jest.spyOn(props, "signInUser");
    const loginWrapped = wrapper.wrap(wrapper.find(LoginForm));

    loginWrapped.instance().handleSubmit({ preventDefault: jest.fn() });
    expect(signInUserMethod).not.toHaveBeenCalled();
  });

  test("should validate required fields", () => {
    expect.assertions(2);

    const EnhancedForm = Form.create()(WrappedLoginForm);
    const wrapper = mount(<EnhancedForm {...props} />);
    const loginWrapped = wrapper.wrap(wrapper.find(LoginForm));

    loginWrapped
      .find("input")
      .get(0)
      .props.onChange({
        target: { value: "bitofproperty@rocks.com" }
      });
    loginWrapped
      .find("input")
      .get(1)
      .props.onChange({
        target: { value: "bitofproperty@rocks.com" }
      });

    const signInUserMethod = jest.spyOn(props, "signInUser");
    const pushMethod = jest.spyOn(props.history, "push");

    loginWrapped.instance().handleSubmit({ preventDefault: jest.fn() });
    expect(signInUserMethod).toHaveBeenCalled();
    props.signInUser({}).then(() => {
      expect(pushMethod).toHaveBeenCalledWith("/");
    });
  });

  test("should validate required fields but fail on signin", () => {
    expect.assertions(3);
    props = {
      ...props,
      signInUser: jest.fn(() => Promise.reject())
    };

    const EnhancedForm = Form.create()(WrappedLoginForm);
    const wrapper = mount(<EnhancedForm {...props} />);
    const loginWrapped = wrapper.wrap(wrapper.find(LoginForm));

    loginWrapped
      .find("input")
      .get(0)
      .props.onChange({
        target: { value: "bitofproperty@rocks.com" }
      });
    loginWrapped
      .find("input")
      .get(1)
      .props.onChange({
        target: { value: "bitofproperty@rocks.com" }
      });

    const signInUserMethod = jest.spyOn(props, "signInUser");
    const pushMethod = jest.spyOn(props.history, "push");

    loginWrapped.instance().handleSubmit({ preventDefault: jest.fn() });
    expect(signInUserMethod).toHaveBeenCalled();
    props.signInUser({}).catch(() => {
      expect(pushMethod).not.toHaveBeenCalled();
      expect(loginWrapped.instance().state.loading).toEqual(false);
    });
  });
});
