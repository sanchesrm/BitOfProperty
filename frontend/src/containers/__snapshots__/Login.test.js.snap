// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component render correctly 1`] = `
<Form(Form(LoginForm))
  history={
    Object {
      "push": [MockFunction],
    }
  }
  signInUser={[MockFunction]}
>
  <Form(LoginForm)
    form={
      Object {
        "getFieldDecorator": [Function],
        "getFieldError": [Function],
        "getFieldInstance": [Function],
        "getFieldProps": [Function],
        "getFieldValue": [Function],
        "getFieldsError": [Function],
        "getFieldsValue": [Function],
        "isFieldTouched": [Function],
        "isFieldValidating": [Function],
        "isFieldsTouched": [Function],
        "isFieldsValidating": [Function],
        "isSubmitting": [Function],
        "resetFields": [Function],
        "setFields": [Function],
        "setFieldsInitialValue": [Function],
        "setFieldsValue": [Function],
        "submit": [Function],
        "validateFields": [Function],
        "validateFieldsAndScroll": [Function],
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    signInUser={[MockFunction]}
  >
    <LoginForm
      form={
        Object {
          "getFieldDecorator": [Function],
          "getFieldError": [Function],
          "getFieldInstance": [Function],
          "getFieldProps": [Function],
          "getFieldValue": [Function],
          "getFieldsError": [Function],
          "getFieldsValue": [Function],
          "isFieldTouched": [Function],
          "isFieldValidating": [Function],
          "isFieldsTouched": [Function],
          "isFieldsValidating": [Function],
          "isSubmitting": [Function],
          "resetFields": [Function],
          "setFields": [Function],
          "setFieldsInitialValue": [Function],
          "setFieldsValue": [Function],
          "submit": [Function],
          "validateFields": [Function],
          "validateFieldsAndScroll": [Function],
        }
      }
      history={
        Object {
          "push": [MockFunction],
        }
      }
      signInUser={[MockFunction]}
    >
      <Adapter
        style={
          Object {
            "padding": "10% 40%",
          }
        }
      >
        <Basic
          prefixCls="ant-layout-content"
          style={
            Object {
              "padding": "10% 40%",
            }
          }
        >
          <div
            className="ant-layout-content"
            style={
              Object {
                "padding": "10% 40%",
              }
            }
          >
            <img
              alt=""
              src="bitofproperty.png"
              style={
                Object {
                  "maxWidth": "300px",
                }
              }
            />
            <Form
              hideRequiredMark={false}
              layout="horizontal"
              onSubmit={[Function]}
              prefixCls="ant-form"
              style={
                Object {
                  "maxWidth": "300px",
                }
              }
            >
              <form
                className="ant-form ant-form-horizontal"
                onSubmit={[Function]}
                style={
                  Object {
                    "maxWidth": "300px",
                  }
                }
              >
                <FormItem
                  colon={true}
                  hasFeedback={false}
                  prefixCls="ant-form"
                >
                  <Row
                    className="ant-form-item"
                    gutter={0}
                  >
                    <div
                      className="ant-row ant-form-item"
                    >
                      <Col
                        className="ant-form-item-control-wrapper"
                        key="wrapper"
                      >
                        <div
                          className="ant-form-item-control-wrapper"
                        >
                          <div
                            className="ant-form-item-control"
                          >
                            <span
                              className="ant-form-item-children"
                            >
                              <Input
                                data-__field={
                                  Object {
                                    "name": "username",
                                  }
                                }
                                data-__meta={
                                  Object {
                                    "name": "username",
                                    "originalProps": Object {
                                      "disabled": false,
                                      "placeholder": "Username",
                                      "prefix": <Icon
                                        type="user"
                                      />,
                                      "prefixCls": "ant-input",
                                      "type": "text",
                                    },
                                    "ref": null,
                                    "rules": Array [
                                      Object {
                                        "message": "Username should be an e-mail!",
                                        "type": "email",
                                      },
                                      Object {
                                        "message": "Please put your username",
                                        "required": true,
                                      },
                                    ],
                                    "trigger": "onChange",
                                    "validate": Array [
                                      Object {
                                        "rules": Array [
                                          Object {
                                            "message": "Username should be an e-mail!",
                                            "type": "email",
                                          },
                                          Object {
                                            "message": "Please put your username",
                                            "required": true,
                                          },
                                        ],
                                        "trigger": Array [
                                          "onChange",
                                        ],
                                      },
                                    ],
                                    "valuePropName": "value",
                                  }
                                }
                                disabled={false}
                                id="username"
                                onChange={[Function]}
                                placeholder="Username"
                                prefix={
                                  <Icon
                                    type="user"
                                  />
                                }
                                prefixCls="ant-input"
                                type="text"
                              >
                                <span
                                  className="ant-input-affix-wrapper"
                                >
                                  <span
                                    className="ant-input-prefix"
                                  >
                                    <Icon
                                      type="user"
                                    >
                                      <i
                                        className="anticon anticon-user"
                                      >
                                        <IconReact
                                          className=""
                                          type="user-o"
                                        >
                                          <svg
                                            aria-hidden="true"
                                            className=""
                                            data-icon="user"
                                            fill="currentColor"
                                            height="1em"
                                            key="svg-user"
                                            viewBox="64 64 896 896"
                                            width="1em"
                                          >
                                            <path
                                              d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                                              key="svg-user-svg-0"
                                            />
                                          </svg>
                                        </IconReact>
                                      </i>
                                    </Icon>
                                  </span>
                                  <input
                                    className="ant-input"
                                    data-__field={
                                      Object {
                                        "name": "username",
                                      }
                                    }
                                    data-__meta={
                                      Object {
                                        "name": "username",
                                        "originalProps": Object {
                                          "disabled": false,
                                          "placeholder": "Username",
                                          "prefix": <Icon
                                            type="user"
                                          />,
                                          "prefixCls": "ant-input",
                                          "type": "text",
                                        },
                                        "ref": null,
                                        "rules": Array [
                                          Object {
                                            "message": "Username should be an e-mail!",
                                            "type": "email",
                                          },
                                          Object {
                                            "message": "Please put your username",
                                            "required": true,
                                          },
                                        ],
                                        "trigger": "onChange",
                                        "validate": Array [
                                          Object {
                                            "rules": Array [
                                              Object {
                                                "message": "Username should be an e-mail!",
                                                "type": "email",
                                              },
                                              Object {
                                                "message": "Please put your username",
                                                "required": true,
                                              },
                                            ],
                                            "trigger": Array [
                                              "onChange",
                                            ],
                                          },
                                        ],
                                        "valuePropName": "value",
                                      }
                                    }
                                    disabled={false}
                                    id="username"
                                    onChange={[Function]}
                                    onKeyDown={[Function]}
                                    placeholder="Username"
                                    style={null}
                                    type="text"
                                    value=""
                                  />
                                </span>
                              </Input>
                            </span>
                            <Animate
                              animation={Object {}}
                              component=""
                              componentProps={Object {}}
                              key="help"
                              onAppear={[Function]}
                              onEnd={[Function]}
                              onEnter={[Function]}
                              onLeave={[Function]}
                              transitionAppear={true}
                              transitionEnter={true}
                              transitionLeave={true}
                              transitionName="show-help"
                            />
                          </div>
                        </div>
                      </Col>
                    </div>
                  </Row>
                </FormItem>
                <FormItem
                  colon={true}
                  hasFeedback={false}
                  prefixCls="ant-form"
                >
                  <Row
                    className="ant-form-item"
                    gutter={0}
                  >
                    <div
                      className="ant-row ant-form-item"
                    >
                      <Col
                        className="ant-form-item-control-wrapper"
                        key="wrapper"
                      >
                        <div
                          className="ant-form-item-control-wrapper"
                        >
                          <div
                            className="ant-form-item-control"
                          >
                            <span
                              className="ant-form-item-children"
                            >
                              <Input
                                data-__field={
                                  Object {
                                    "name": "password",
                                  }
                                }
                                data-__meta={
                                  Object {
                                    "name": "password",
                                    "originalProps": Object {
                                      "disabled": false,
                                      "placeholder": "Password",
                                      "prefix": <Icon
                                        type="lock"
                                      />,
                                      "prefixCls": "ant-input",
                                      "type": "password",
                                    },
                                    "ref": null,
                                    "rules": Array [
                                      Object {
                                        "message": "Please input your Password!",
                                        "required": true,
                                      },
                                    ],
                                    "trigger": "onChange",
                                    "validate": Array [
                                      Object {
                                        "rules": Array [
                                          Object {
                                            "message": "Please input your Password!",
                                            "required": true,
                                          },
                                        ],
                                        "trigger": Array [
                                          "onChange",
                                        ],
                                      },
                                    ],
                                    "valuePropName": "value",
                                  }
                                }
                                disabled={false}
                                id="password"
                                onChange={[Function]}
                                placeholder="Password"
                                prefix={
                                  <Icon
                                    type="lock"
                                  />
                                }
                                prefixCls="ant-input"
                                type="password"
                              >
                                <span
                                  className="ant-input-affix-wrapper"
                                >
                                  <span
                                    className="ant-input-prefix"
                                  >
                                    <Icon
                                      type="lock"
                                    >
                                      <i
                                        className="anticon anticon-lock"
                                      >
                                        <IconReact
                                          className=""
                                          type="lock-o"
                                        >
                                          <svg
                                            aria-hidden="true"
                                            className=""
                                            data-icon="lock"
                                            fill="currentColor"
                                            height="1em"
                                            key="svg-lock"
                                            viewBox="64 64 896 896"
                                            width="1em"
                                          >
                                            <path
                                              d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0z"
                                              key="svg-lock-svg-0"
                                            />
                                          </svg>
                                        </IconReact>
                                      </i>
                                    </Icon>
                                  </span>
                                  <input
                                    className="ant-input"
                                    data-__field={
                                      Object {
                                        "name": "password",
                                      }
                                    }
                                    data-__meta={
                                      Object {
                                        "name": "password",
                                        "originalProps": Object {
                                          "disabled": false,
                                          "placeholder": "Password",
                                          "prefix": <Icon
                                            type="lock"
                                          />,
                                          "prefixCls": "ant-input",
                                          "type": "password",
                                        },
                                        "ref": null,
                                        "rules": Array [
                                          Object {
                                            "message": "Please input your Password!",
                                            "required": true,
                                          },
                                        ],
                                        "trigger": "onChange",
                                        "validate": Array [
                                          Object {
                                            "rules": Array [
                                              Object {
                                                "message": "Please input your Password!",
                                                "required": true,
                                              },
                                            ],
                                            "trigger": Array [
                                              "onChange",
                                            ],
                                          },
                                        ],
                                        "valuePropName": "value",
                                      }
                                    }
                                    disabled={false}
                                    id="password"
                                    onChange={[Function]}
                                    onKeyDown={[Function]}
                                    placeholder="Password"
                                    style={null}
                                    type="password"
                                    value=""
                                  />
                                </span>
                              </Input>
                            </span>
                            <Animate
                              animation={Object {}}
                              component=""
                              componentProps={Object {}}
                              key="help"
                              onAppear={[Function]}
                              onEnd={[Function]}
                              onEnter={[Function]}
                              onLeave={[Function]}
                              transitionAppear={true}
                              transitionEnter={true}
                              transitionLeave={true}
                              transitionName="show-help"
                            />
                          </div>
                        </div>
                      </Col>
                    </div>
                  </Row>
                </FormItem>
                <FormItem
                  colon={true}
                  hasFeedback={false}
                  prefixCls="ant-form"
                >
                  <Row
                    className="ant-form-item"
                    gutter={0}
                  >
                    <div
                      className="ant-row ant-form-item"
                    >
                      <Col
                        className="ant-form-item-control-wrapper"
                        key="wrapper"
                      >
                        <div
                          className="ant-form-item-control-wrapper"
                        >
                          <div
                            className="ant-form-item-control"
                          >
                            <span
                              className="ant-form-item-children"
                            >
                              <Button
                                block={false}
                                ghost={false}
                                htmlType="submit"
                                loading={false}
                                prefixCls="ant-btn"
                                style={
                                  Object {
                                    "width": "100%",
                                  }
                                }
                                type="primary"
                              >
                                <Wave>
                                  <button
                                    className="ant-btn ant-btn-primary"
                                    onClick={[Function]}
                                    style={
                                      Object {
                                        "width": "100%",
                                      }
                                    }
                                    type="submit"
                                  >
                                    <span
                                      key=".0"
                                    >
                                      Log in
                                    </span>
                                  </button>
                                </Wave>
                              </Button>
                            </span>
                            <Animate
                              animation={Object {}}
                              component=""
                              componentProps={Object {}}
                              key="help"
                              onAppear={[Function]}
                              onEnd={[Function]}
                              onEnter={[Function]}
                              onLeave={[Function]}
                              transitionAppear={true}
                              transitionEnter={true}
                              transitionLeave={true}
                              transitionName="show-help"
                            />
                          </div>
                        </div>
                      </Col>
                    </div>
                  </Row>
                </FormItem>
              </form>
            </Form>
          </div>
        </Basic>
      </Adapter>
    </LoginForm>
  </Form(LoginForm)>
</Form(Form(LoginForm))>
`;
